### PR TITLE
parse_tldrlist: propagate opendir fails instead of segfaulting

### PR DESCRIPTION
**Pull request by pixelcmtd**
## What does it do?
Return 1, when `opendir` fails.

## Why the change?
Fixes segfaults.

## How can this be tested?
Reproduce #71, or `chown root:root ~/.tldrc && chmod 700 ~/.tldrc`.

## Relevant tickets?
While not fixing it, this does stop #71 from segfaulting.

### CODE REVIEW

File: src/parser.c [hunk 1]
```diff
@@ -268,12 +268,14 @@ parse_tldrlist(char const *path, char const *platform)
     if (sstrncat(fullpath, &len, STRBUFSIZ, platform, strlen(platform)))
         return 1;
 
+    directory = opendir(fullpath);
+    if (directory == NULL)
+        return 1;
```
Comment (MasterOdin) [inline src/parser.c:273]: For this return (and the ones above), is the user given an appropriate error message, or does it just silently exit with error code `1`?

Comment (pixelcmtd) [reply src/parser.c:273]: It actually currently silently exits with error code `0`, which is of course totally wrong.

Comment (MasterOdin) [reply src/parser.c:273]: Hm, that's suprising as this is bubbling up to

https://github.com/tldr-pages/tldr-c-client/blob/2adb81b32978dacc2ddd6e0f487d6848a215cb8c/src/tldr.c#L148-L149

which should then returning `EXIT_FAILURE` to the caller?

Comment (MasterOdin) [reply src/parser.c:273]: Ah, gotcha. Would you be able to just add a `fprintf(stderr, ...)` output here complaining something about the directory path?

**Review by MasterOdin — APPROVED** [review]

**Pull request merged**